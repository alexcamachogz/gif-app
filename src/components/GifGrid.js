import React from 'react'
import { useFetchGif } from '../hooks/useFetchGif'
import { GifGridItem } from './GifGridItem'
// import PropTypes from 'prop-types'

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGif(category)
  return (
    <>
      <h3>{category}</h3>
      {loading && <p>Loading...</p>}
      <div className="card-grid">
        {
          images.map(img => (
            <GifGridItem key={img.id} {...img}/>
          ))
        }
      </div>
    </>
  )
}