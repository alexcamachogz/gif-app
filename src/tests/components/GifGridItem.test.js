import React from 'react'
import { shallow } from 'enzyme'
import { GifGridItem } from '../../components/GifGridItem'

describe('Testing <GifGridItem />', () => {

  const title = 'Gif title'
  const url = 'https://localhost/image.jpg'
  const wrapper = shallow(<GifGridItem title={title} url={url}/>)

  test('Should be show <GifGridItem /> properly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('Should have a paragraph with the title', () => {
    const p = wrapper.find('p')
    expect((p.text().trim())).toBe(title)
  })

  test('Should have an image with url and alt', () => {
    const img = wrapper.find('img')
    expect(img.prop('src')).toBe(url)
    expect(img.prop('alt')).toBe(title)
  })

  test('Should have animate__fadeIn as a class', () => {
    const div = wrapper.find('div')
    const isAnimate = div.prop('className').includes('animate__fadeIn')
    expect(isAnimate).toBe(true)
  })
})