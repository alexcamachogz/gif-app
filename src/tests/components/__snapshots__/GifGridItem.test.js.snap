// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifGridItem /> Should be show <GifGridItem /> properly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Gif title"
    src="https://localhost/image.jpg"
  />
  <p>
    Gif title
  </p>
</div>
`;
